discretisedfield/field.py

```diff
@@ -3475,3 +3476,115 @@ def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
         else:
             return self.__class__(mesh[0], dim=result.shape[-1], value=result,
                                   components=self.components)
+
+    def to_xarray(self, name='field', units=None):
+        """Field value as ``xarray.DataArray``.
+
+        The function returns an ``xarray.DataArray`` with dimensions ``x``,
+        ``y``, ``z``, and ``comp`` (``only if field.dim > 1``). The coordinates
+        of the geometric dimensions are derived from ``self.mesh.axis_points``,
+        and for vector field components from ``self.components``. The ``units``
+        attribute of geometric dimensions is set to
+        ``self.mesh.attributes['unit']``.
+
+        The name and units of the field ``DataArray`` can be set by passing
+        ``name`` and ``units``. If the type of value passed to any of the two
+        arguments is not ``str``, then a ``TypeError`` is raised.
+
+        Parameters
+        ----------
+        name : str, optional
+
+            String to set name of the field ``DataArray``.
+
+        units : str, optional
+
+            String to set units of the field ``DataArray``.
+
+        Returns
+        -------
+        xarray.DataArray
+
+            Field values DataArray.
+
+        Raises
+        ------
+        TypeError
+
+            If either ``name`` or ``units`` argument is not a string.
+
+        Examples
+        --------
+        >>> import discretisedfield as df
+        ...
+        >>> p1 = (0, 0, 0)
+        >>> p2 = (10, 10, 10)
+        >>> cell = (1, 1, 1)
+        >>> mesh = df.Mesh(p1=p1, p2=p2, cell=cell)
+        >>> field = df.Field(mesh=mesh, dim=3, value=(1, 0, 0), norm=1.)
+        ...
+        >>> field
+        Field(...)
+        >>> field.to_xarray()
+        <xarray.DataArray 'field' (x: 10, y: 10, z: 10, comp: 3)>
+        ...
+        Coordinates:
+          * x        (x) float64 0.5 1.5 2.5 3.5 4.5 5.5 6.5 7.5 8.5 9.5
+          * y        (y) float64 0.5 1.5 2.5 3.5 4.5 5.5 6.5 7.5 8.5 9.5
+          * z        (z) float64 0.5 1.5 2.5 3.5 4.5 5.5 6.5 7.5 8.5 9.5
+          * comp     (comp) <U1 'x' 'y' 'z'
+        >>> xa = field.to_xarray()
+        >>> xa.sel(comp='x')
+        <xarray.DataArray 'field' (x: 10, y: 10, z: 10)>
+        ...
+        Coordinates:
+          * x        (x) float64 0.5 1.5 2.5 3.5 4.5 5.5 6.5 7.5 8.5 9.5
+          * y        (y) float64 0.5 1.5 2.5 3.5 4.5 5.5 6.5 7.5 8.5 9.5
+          * z        (z) float64 0.5 1.5 2.5 3.5 4.5 5.5 6.5 7.5 8.5 9.5
+            comp     <U1 'x'
+
+        """
+        if not isinstance(name, str):
+            msg = "Name argument must be a string."
+            raise TypeError(msg)
+
+        if units is not None and not isinstance(units, str):
```

Comment:
Is just `if not isinstance(units, str)` a sufficient condition?

Reply:
The default value of `units` is `None`. `if not isinstance(units, str)` will raise `TypeError` for `units=None`

Reply:
Something like `isinstance(None, str)` [-> `False`] should work just fine.

Reply:
```Python
if not isinstance(units, str):
            msg = "Units argument must be a string."
            raise TypeError(msg)
```
@lang-m  Sorry, I did not get what you mean. The code above raises `TypeError` if I try `field.to_xarray()` (without the `units` argument). Am I missing something obvious?

Reply:
@swapneelap The code *should raise* a `TypeError` if you do not pass `units` (i.e. the default `units = None`) because you manually raise it in the last line (the message would be important but it will for sure be what you manually set for `msg`).

Reply:
The actual issue is slightly different (I missed this in the beginning). We want `units` to be optional (but of type `str` if passed). So we need to keep the current implementation (only raising if not `None` and not `str`).